packages/apollo-server-core/src/types.ts

```diff
@@ -87,6 +93,12 @@ export interface GraphQLService {
     engine?: GraphQLServiceEngineConfig;
   }): Promise<GraphQLServiceConfig>;
   onSchemaChange(callback: SchemaChangeCallback): Unsubscriber;
+  executor<TContext>(
```

Comment:
Dogearing this line for us to revisit at some point. I'm not sure this behaves as expected based on past experience with type variables on properties of interfaces (what a mouthful).

Reply:
Unclear if it's misbehaving, but I've preserved this concern outside the context of this PR via a comment: 81a72e237460e6a8bf2912cf0b6436ef7f0dc9d3

Reply:
Perfect!